crates/graphql_artifact_generation/src/reader_ast.rs

```diff
@@ -37,7 +37,7 @@ fn generate_reader_ast_node(
                         indentation_level,
                         initial_variable_context,
                     ),
-                    FieldType::ClientField(client_field_id) => {
+                    FieldType::ClientField(ClientType::ClientField(client_field_id)) => {
```

Comment:
(1) hm... we'd have to panic here, since a scalar field cannot be a client pointer. I think in `validate_schema`, it doesn't make sense for scalar field selections to be `ClientType`. instead, it makes sense for linked fields to have that enum.